packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/VideoRenderer.java

```diff
@@ -0,0 +1,322 @@
+          + "            {\n"
+          + "                vec4 c = texture2D(texSampler, varUvs);\n"
+          + "                gl_FragColor = vec4(c.r, c.g, c.b, c.a);\n"
+          + "            }";
+
+  private final int[] textureHandles = new int[1];
+
+  private final float[] vertices =
+      new float[] {
+        -1.0f, -1.0f, 0.0f, 0f, 0f, -1.0f, 1.0f, 0.0f, 0f, 1f, 1.0f, 1.0f, 0.0f, 1f, 1f, 1.0f,
+        -1.0f, 0.0f, 1f, 0f
+      };
+
+  private final int[] indices = new int[] {2, 1, 0, 0, 3, 2};
+
+  private int program;
+  private int vertexHandle = 0;
+  private final int[] bufferHandles = new int[2];
+  private int uvsHandle = 0;
+  private int texMatrixHandle = 0;
+  private int mvpHandle = 0;
+
+  EGLDisplay display;
+  EGLContext context;
+  EGLSurface surface;
+  private Thread thread;
+  private final Surface outputSurface;
+  private SurfaceTexture inputSurfaceTexture;
+  private Surface inputSurface;
+
+  private HandlerThread surfaceTextureFrameAvailableHandler;
+  private final Object surfaceTextureAvailableFrameLock = new Object();
+  private Boolean surfaceTextureFrameAvailable = false;
+
+  private final int recordingWidth;
+  private final int recordingHeight;
+  private int rotation = 0;
+
+  private final Object lock = new Object();
+
+  /** Gets surface for input. Blocks until surface is ready. */
+  public Surface getInputSurface() throws InterruptedException {
+    synchronized (lock) {
+      while (inputSurface == null) {
+        lock.wait();
+      }
+    }
+    return inputSurface;
+  }
+
+  public VideoRenderer(Surface outputSurface, int recordingWidth, int recordingHeight) {
+    this.outputSurface = outputSurface;
+    this.recordingHeight = recordingHeight;
+    this.recordingWidth = recordingWidth;
+    startOpenGL();
+    Log.d(TAG, "VideoRenderer setup complete");
+  }
+
+  /** Stop rendering and cleanup resources. */
+  public void close() {
+    thread.interrupt();
+    surfaceTextureFrameAvailableHandler.quitSafely();
+    inputSurfaceTexture.release();
+  }
+
+  /** Configures openGL. Must be called in same thread as draw is called. */
+  private void configureOpenGL() {
+    synchronized (lock) {
+      display = EGL14.eglGetDisplay(EGL14.EGL_DEFAULT_DISPLAY);
+      if (display == EGL14.EGL_NO_DISPLAY)
+        throw new RuntimeException(
+            "eglDisplay == EGL14.EGL_NO_DISPLAY: "
+                + GLUtils.getEGLErrorString(EGL14.eglGetError()));
+
+      int[] version = new int[2];
+      if (!EGL14.eglInitialize(display, version, 0, version, 1))
+        throw new RuntimeException(
+            "eglInitialize(): " + GLUtils.getEGLErrorString(EGL14.eglGetError()));
+
+      int[] attribList =
+          new int[] {
+            EGL14.EGL_RED_SIZE, 8,
+            EGL14.EGL_GREEN_SIZE, 8,
+            EGL14.EGL_BLUE_SIZE, 8,
+            EGL14.EGL_ALPHA_SIZE, 8,
+            EGL14.EGL_RENDERABLE_TYPE, EGL14.EGL_OPENGL_ES2_BIT,
+            EGLExt.EGL_RECORDABLE_ANDROID, 1,
```

Comment:
We could disable the plugin or just disable switching the camera mid recording for any device below API 26. What do you think?

Otherwise if we left it, I think it will only fail on a subset of devices below API 26 when 'eglSwapBuffers' is called.